core/src/main/java/com/linecorp/armeria/common/stream/DeferredStreamMessage.java

```diff
@@ -81,10 +93,19 @@ protected void delegate(StreamMessage<T> delegate) {
             }).exceptionally(CompletionActions::log);
         }
 
-        @SuppressWarnings("unchecked")
-        final PendingSubscription<? super T> ps = pendingSubscription;
-        if (ps != null) {
-            subscribeToDelegate(delegate, ps);
+        // There are 4 state in class.
+        // If delegate is called first, then it will match INIT -> DELEGATED.
+        // If subscribe is called first, than it will match SUBSCRIBED -> SUBSCRIBED_AND_DELEGATED.
+        // It's not possible to enter this for loop with DELEGATED/SUBSCRIBED_AND_DELEGATED state,
+        // because this method doesn't allow to set delegate twice.
+        for (;;) {
```

Comment:
What if this delegate() is called for a while after subscribe() is called?? Then, the thread will be consumed in the loop and cannot do anything.

Reply:
Oops~ my mistake forget about that~